packages/account/src/Sections/Profile/PhoneVerification/cancel-phone-verification-modal.tsx

```diff
@@ -1,63 +1,81 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { Modal, Text } from '@deriv-com/quill-ui';
 import { Localize } from '@deriv/translations';
-import { useHistory } from 'react-router';
+import { useHistory, useLocation } from 'react-router';
 import { observer, useStore } from '@deriv/stores';
 import { LabelPairedCircleXmarkLgRegularIcon } from '@deriv/quill-icons';
 import { useDevice } from '@deriv-com/ui';
-import { routes } from '@deriv/shared';
+import { usePhoneVerificationAnalytics } from '@deriv/hooks';
 
-type TCancelPhoneVerificationModal = {
-    should_show_cancel_verification_modal: boolean;
-    setShouldShowCancelVerificationModal: (value: boolean) => void;
-};
+const CancelPhoneVerificationModal = observer(() => {
+    const history = useHistory();
+    const location = useLocation();
+    const [show_modal, setShowModal] = useState(false);
+    const [next_location, setNextLocation] = useState(location.pathname);
+    const { ui, client } = useStore();
+    const { setShouldShowPhoneNumberOTP, is_forced_to_redirect } = ui;
```

Comment:
is_forced_to_redirect is too generic name 
we need to have a variable name specific to phone number verification flow. 
or is it possisble to create a  provider and wrao all Phone number flow init, so no need to add in UI store

Reply:
I'll do renaming for now. will create a PR to change it to Context 🙏